coralproject/talk
# Check if token is in localStorage before logout

## What does this PR do?

We are using an external auth platform, so we need to logout user on talk after the user is logged out on my auth system. The problem is that there is no way to, externally, check if the user is already logged on Talk and if we call TalkEmbed.logout() and the user is not authenticated, it raises an error on the client and server, printing a lot of error messages on the talk log.

So we thought that doesn't make sense calling the backend and doing all logout steps if there is no token in local storage. We made the change in this PR.

I would like to now your opinion about this change, and also how could we write an automated test for this.

## How do I test this PR?

- open debugger
- call TalkEmbed.logout()
- call TalkEmbed.logout() again
- should NOT raise error on server and client

**Before PR:**
![screen shot 2018-04-13 at 14 47 35](https://user-images.githubusercontent.com/25023177/38751467-994caedc-3f2e-11e8-81d5-93650dede29d.png)

![screen shot 2018-04-13 at 15 27 15](https://user-images.githubusercontent.com/25023177/38751702-394d2952-3f2f-11e8-96ee-b7b5e0a2b386.png)



**After PR:**
![screen shot 2018-04-13 at 14 48 16](https://user-images.githubusercontent.com/25023177/38751468-996cb632-3f2e-11e8-98b8-92263426d0be.png)
